tpm2/Makefile

```diff
@@ -100,25 +124,30 @@ pre:
 	rm -f include/*.h~
 	mkdir -p $(BIN_DIR)
 	mkdir -p $(OBJECT_DIRS)
-	mkdir -p $(DOC_DIR)
+	mkdir -p $(DOC_DIRS)
+	mkdir -p $(LOGGER_LIB_DIR)
 
-doc: $(HEADER_FILES) $(UTIL_SOURCES) $(TOOL_SOURCES) $(SEAL_SOURCES) $(UNSEAL_SOURCES) $(GETKEY_SOURCES) Doxyfile
+doc: $(HEADER_FILES) $(UTIL_SOURCES) $(TOOL_SOURCES) $(SEAL_SOURCES) $(UNSEAL_SOURCES) $(GETKEY_SOURCES) $(LOGGER_HEADERS) $(LOGGER_SOURCES) Doxyfile
 	doxygen Doxyfile
 
+$(LOGGER_OBJECTS): $(LOGGER_SOURCES) $(LOGER_HEADERS)
+	$(CC) $(LOGGER_CFLAGS) -I$(LOGGER_INC_DIR) $< -o $@
+
 $(SEAL_OBJ_DIR)/%.o: $(SEAL_SRC_DIR)/%.c | $(SEAL_OBJ_DIR) 
```

Comment:
Do we want to be able to build the library objects without having to build the executables?